### DISCUSSION TITLE
How to set the sensor position in a manully defined rectangular mesh for ERT inversion?

### DISCUSSION THREAD

**Discussion by lwang19-ai** [gimli-org/pyGIMLi#601] [Q&A] [answered]
I want to do the ERT inversion with my own defined rectangular mesh. Real measurement data is used. What should I do for the sensor positions?
In the example, an instruction is: " Due to experience, it's convenient to add further refinement nodes in a distance of 10% of electrode spacing to achieve sufficient numerical accuracy." Then the code:
` geom.createNode(p - [0, 0.1])`
The mesh is then automatically refined by a triangular mesh. When the mesh is manually defined as rectangular, how should we get enough numerical accuracy? Do I need to do the same and also use 0.1 for example as above? Or once the sensor positions are contained by mesh nodes, it's enough for inversion?

**Comment by lwang19-ai** [comment]
I get very strange results like this:
![image](https://github.com/gimli-org/gimli/assets/57149104/b064c8a0-5fc6-4eb1-9a92-6d341e484db2)
This mesh is generated following the question here: [2D Rectangular mesh with geometry surface](https://github.com/gimli-org/gimli/issues/589)
I am wondering what's wrong here. Why do I get the same results as columns?

**Comment by halbmy** [comment, marked as answer]
A geometry (PLC) is a geometric description that is eventually used to create a mesh. If you want a rectangular mesh, you get a mesh straight away by `mt.createGrid(x, y)`. So if you need to have sensors as nodes, you could make sure the positions are contained in the x and y vectors. In your case it seems like you just want to add topography, e.g. by
```
grid = mt.createGrid(x, y)
mt = np.interp(pg.x(grid), tx, ty)  # topography contained in tx/ty pairs
for i, n in enumerate(grid.nodes()):
    n.setPosition([n.x(), n.y()+mt[i]])
```
`mt.createGrid` could take a `topography` keyword so that it is done automatically.
